cpp/src/arrow/compute/kernels/scalar_fill_null.cc

```diff
@@ -84,6 +84,52 @@ struct FillNullFunctor<Type, enable_if_t<is_number_type<Type>::value>> {
   }
 };
 
+template <typename Type>
+struct FillNullFunctor<Type, enable_if_t<is_base_binary_type<Type>::value>> {
+  using BuilderType = typename TypeTraits<Type>::BuilderType;
+
+  static void Exec(KernelContext* ctx, const ExecBatch& batch, Datum* out) {
+    const ArrayData& input = *batch[0].array();
+    const auto& fill_value_scalar =
+        checked_cast<const BaseBinaryScalar&>(*batch[1].scalar());
+    util::string_view fill_value =
+        static_cast<util::string_view>(*fill_value_scalar.value);
```

Comment:
The `static_cast` looks weird IMHO. How about using a constructor call?

Reply:
Totally bizar, it's a copy-paste from some previous code of mine, will also change that.